app/models/manageiq/providers/amazon/storage_manager/ebs/refresh_parser_inventory_object.rb

```diff
@@ -59,6 +63,53 @@ def parse_snapshot(snap)
     }
   end
 
+  def link_volume_to_disk(volume_hash, attachments)
+    uid = volume_hash[:ems_ref]
+
+    attachments.each do |a|
+      if a['device'].blank?
+        _log.warn "#{log_header}: Volume: #{uid}, is missing a mountpoint, skipping the volume processing"
+        _log.warn "#{log_header}:   EMS: #{@ems.name}, Instance: #{a['instance_id']}"
+        next
+      end
+
+      dev = File.basename(a['device'])
+
+      vm = inventory_collections[:vms].find(a['instance_id']).data
+      unless vm
+        _log.warn "VM referenced by attachment (#{a['instance_id']} not found."
+        next
+      end
+
+      hardware = vm.hardware
```

Comment:
I get something for the VM, but definitely not the correct thing. Not sure how to get there. You can see in the `refresh_parser` what's needed: https://github.com/ManageIQ/manageiq-providers-amazon/pull/125/files#diff-9ed63d88e4d31be7b6b921a61efec9fdR100